oap-server/server-library/library-client/src/test/java/org/apache/skywalking/oap/server/library/client/elasticsearch/ITElasticSearchClient.java

```diff
@@ -51,7 +52,10 @@
     @Before
     public void before() throws IOException {
         final String esAddress = System.getProperty("elastic.search.address");
-        client = new ElasticSearchClient(esAddress, "", "test", "test");
+        final String esNamespace = System.getProperty("elastic.search.namespace", "");
+        final String esUser = System.getProperty("elastic.search.user", "test");
+        final String esPassword = System.getProperty("elastic.search.password", "test");
```

Comment:
One question, who will set these properties?

Reply:
If not, then,  namespace feature hasn't been tested.

Reply:
Can I use the ```maven-failsafe-plugin```, as follows
```
                    <plugin>
                        <groupId>org.apache.maven.plugins</groupId>
                        <artifactId>maven-failsafe-plugin</artifactId>
                        <configuration>
                            <systemPropertyVariables>
                                <elastic.search.address>
                                    ${docker.hostname}:${es-port}
                                </elastic.search.address>
                                <elastic.search.namespace>
                                    test
                                </elastic.search.namespace>
                            </systemPropertyVariables>
                        </configuration>
                        <executions>
                            <execution>
                                <goals>
                                    <goal>integration-test</goal>
                                    <goal>verify</goal>
                                </goals>
                            </execution>
                        </executions>
                    </plugin>
```

Reply:
After adding ```namespace```,```getIndex``` has the same problem, which I will modify together
https://github.com/apache/skywalking/blob/b21db17148df12f34a13a2b1693e009e11e8d7a2/oap-server/server-library/library-client/src/main/java/org/apache/skywalking/oap/server/library/client/elasticsearch/ElasticSearchClient.java#L138-L146

Reply:
System properties are used for uncertain variables, because of docker env. You don't need this, actually, you could add it to test codes directly, which will be more clear.

Reply:
Also, @Arugal `#getIndex` is not used in codes anymore, we only use it in IT test, let's move it to ITElasticSearchClient? That will make us has fewer codes concern of index name.

Reply:
> Also, @Arugal `#getIndex` is not used in codes anymore, we only use it in IT test, let's move it to ITElasticSearchClient? That will make us has fewer codes concern of index name.

```#getIndex``` needs to invoker ```RestHighLevelClient#getLowLevelClient``` ,unless we can get a ```RestHighLevelClient``` instance via ```ElasticSearchClient```

Reply:
> System properties are used for uncertain variables, because of docker env. You don't need this, actually, you could add it to test codes directly, which will be more clear.

done

Reply:
> > Also, @Arugal `#getIndex` is not used in codes anymore, we only use it in IT test, let's move it to ITElasticSearchClient? That will make us has fewer codes concern of index name.
> 
> `#getIndex` needs to invoker `RestHighLevelClient#getLowLevelClient` ,unless we can get a `RestHighLevelClient` instance via `ElasticSearchClient`



> getIndex method is still there. I think no ine uses it, should remove, right?

Can I do what I said above?